docker/docker_client.go

```diff
@@ -215,6 +218,18 @@ func newDockerClientFromRef(sys *types.SystemContext, ref dockerReference, write
 	client.signatureBase = sigBase
 	client.scope.actions = actions
 	client.scope.remoteName = reference.Path(ref.ref)
+
+	// Adapt the scope to make authentication work when a prefix is provided
+	if client.registry.Prefix != "" && client.registry.Prefix != client.registry.URL {
```

Comment:
This code looks quite complex. Please add verbose comments to make sure the code matches the intention and to ease reading.

Reply:
This is removed in my latest appraoch.